core/domain/exp_jobs_one_off_test.py

```diff
@@ -766,9 +1455,25 @@ def test_that_mapreduce_job_works(self):
         self.assertEqual(mapping.largest_state_id_used, 2)
         self.assertDictEqual(mapping.state_names_to_ids, expected_mapping)
 
+    def test_no_action_is_performed_for_deleted_exploration(self):
+        """Test that no action is performed on deleted explorations."""
 
-class ExplorationContentValidationJobForTextAngularTests(
-        test_utils.GenericTestBase):
+        self.save_new_valid_exploration(self.EXP_ID, self.owner_id)
+
+        exp_services.delete_exploration(self.owner_id, self.EXP_ID)
+
+        # Start ExplorationStateIdMapping job on published exploration.
+        job_id = exp_jobs_one_off.ExplorationStateIdMappingJob.create_new()
+        exp_jobs_one_off.ExplorationStateIdMappingJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        self.assertEqual(
+            exp_jobs_one_off.ExplorationStateIdMappingJob.get_output(job_id),
+            []
+        )
+
+
+class HintsAuditOneOffJobTests(test_utils.GenericTestBase):
```

Comment:
Here also, should we test multiple exps are handled correctly?

Reply:
Done.